[1.0] Platform mDNS no longer builds on Linux

### Reproduction steps

- Run `./scripts/build_python.sh -m platform -i separate`
- Build fails

```
Logs:
In file included from ../../src/inet/IPAddress.h:36,
                 from ../../src/lib/dnssd/platform/Dnssd.h:31,
                 from ../../src/platform/Linux/DnssdImpl.h:37,
                 from ../../src/platform/Linux/DnssdImpl.cpp:18:
../../src/platform/Linux/DnssdImpl.cpp: In static member function 'static void chip::Dnssd::MdnsAvahi::HandleBrowse(AvahiServiceBrowser*, AvahiIfIndex, AvahiProtocol, AvahiBrowserEvent, const char*, const char*, const char*, AvahiLookupResultFlags, void*)':
../../src/lib/core/CHIPError.h:152:14: error: cannot convert 'chip::ChipError' to 'bool' in argument passing
  152 |     (::chip::ChipError(::chip::ChipError::SdkErrorConstant<(part), (code)>::value, __FILE__, __LINE__))
      |     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |              |
      |              chip::ChipError
../../src/lib/core/CHIPError.h:401:28: note: in expansion of macro 'CHIP_SDK_ERROR'
  401 | #define CHIP_CORE_ERROR(e) CHIP_SDK_ERROR(::chip::ChipError::SdkPart::kCore, (e))
      |                            ^~~~~~~~~~~~~~
../../src/lib/core/CHIPError.h:1980:64: note: in expansion of macro 'CHIP_CORE_ERROR'
 1980 | #define CHIP_ERROR_INTERNAL                                    CHIP_CORE_ERROR(0xac)
      |                                                                ^~~~~~~~~~~~~~~
../../src/platform/Linux/DnssdImpl.cpp:619:59: note: in expansion of macro 'CHIP_ERROR_INTERNAL'
  619 |         context->mCallback(context->mContext, nullptr, 0, CHIP_ERROR_INTERNAL);
      |                                                           ^~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics
```


### Bug prevalence

100%

### GitHub hash of the SDK that was being used

e4c075f90da9512993fa2003043df0237e5d7f34

### Platform

core

### Platform Version(s)

_No response_

### Type

Platform Issue

### Anything else?

Since https://github.com/project-chip/connectedhomeip/pull/19554 got erroneously merged, Avahi-based DnsSd on Linux no longer builds

The error is that a new argument was added(bool finalBrowse) to:
```
using DnssdBrowseCallback = void (*)(void * context, DnssdService * services, size_t servicesSize, bool finalBrowse,
                                     CHIP_ERROR error);
```

However **not all callsites** were updated!

--- a/.github/workflows/examples-linux-standalone.yaml
+++ b/.github/workflows/examples-linux-standalone.yaml
@@ -76,7 +76,7 @@ jobs:
               timeout-minutes: 10
               run: |
                   ./scripts/run_in_build_env.sh \
-                  "./scripts/build/build_examples.py --no-log-timestamps --target address-resolve-tool-platform-mdns-ipv6only build"
+                  "./scripts/build/build_examples.py --no-log-timestamps --target linux-x64-address-resolve-tool-platform-mdns-ipv6only build"
             - name: Build example Standalone Echo Client
               timeout-minutes: 10
               run: |

--- a/.github/workflows/examples-linux-standalone.yaml
+++ b/.github/workflows/examples-linux-standalone.yaml
@@ -25,7 +25,7 @@ concurrency:
 jobs:
     linux_standalone:
         name: Linux Standalone
-        timeout-minutes: 70
+        timeout-minutes: 90
 
         env:
             BUILD_TYPE: gn_linux
@@ -72,6 +72,11 @@ jobs:
               run: |
                   ./scripts/run_in_build_env.sh \
                   "./scripts/build/build_examples.py --no-log-timestamps --target-glob '*-chip-cert' build"
+            - name: Build minmdns example with platform dns
+              timeout-minutes: 10
+              run: |
+                  ./scripts/run_in_build_env.sh \
+                  "./scripts/build/build_examples.py --no-log-timestamps --target address-resolve-tool-platform-mdns-ipv6only build"
             - name: Build example Standalone Echo Client
               timeout-minutes: 10
               run: |

--- a/src/platform/Linux/DnssdImpl.cpp
+++ b/src/platform/Linux/DnssdImpl.cpp
@@ -616,7 +616,7 @@ void MdnsAvahi::HandleBrowse(AvahiServiceBrowser * browser, AvahiIfIndex interfa
     switch (event)
     {
     case AVAHI_BROWSER_FAILURE:
-        context->mCallback(context->mContext, nullptr, 0, CHIP_ERROR_INTERNAL);
+        context->mCallback(context->mContext, nullptr, 0, true, CHIP_ERROR_INTERNAL);
         avahi_service_browser_free(browser);
         chip::Platform::Delete(context);
         break;

